src/main/java/marquez/db/DatasourceDao.java

```diff
@@ -17,23 +17,28 @@
 import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
-import marquez.db.mappers.DataSourceRowMapper;
-import marquez.db.models.DataSourceRow;
+import marquez.common.models.DatasourceUrn;
+import marquez.db.mappers.DatasourceRowMapper;
+import marquez.db.models.DatasourceRow;
 import org.jdbi.v3.sqlobject.config.RegisterRowMapper;
+import org.jdbi.v3.sqlobject.customizer.Bind;
 import org.jdbi.v3.sqlobject.customizer.BindBean;
 import org.jdbi.v3.sqlobject.statement.SqlQuery;
 import org.jdbi.v3.sqlobject.statement.SqlUpdate;
 
-@RegisterRowMapper(DataSourceRowMapper.class)
-public interface DataSourceDao {
+@RegisterRowMapper(DatasourceRowMapper.class)
+public interface DatasourceDao {
   @SqlUpdate(
-      "INSERT INTO datasources (guid, name, connection_url) "
-          + "VALUES (:uuid, :name, :connectionUrl)")
-  void insert(@BindBean DataSourceRow dataSourceRow);
+      "INSERT INTO datasources (guid, urn, name, connection_url) "
+          + "VALUES (:uuid, :urn, :name, :connectionUrl)")
+  void insert(@BindBean DatasourceRow datasourceRow);
 
   @SqlQuery("SELECT * FROM datasources WHERE guid = :uuid")
-  Optional<DataSourceRow> findBy(UUID uuid);
+  Optional<DatasourceRow> findBy(@Bind("uuid") UUID uuid);
+
+  @SqlQuery("SELECT * FROM datasources WHERE urn = :urn")
+  Optional<DatasourceRow> findBy(@Bind("urn") DatasourceUrn urn);
 
   @SqlQuery("SELECT * FROM datasources LIMIT :limit OFFSET :offset")
-  List<DataSourceRow> findAll(Integer limit, Integer offset);
+  List<DatasourceRow> findAll(@Bind("limit") Integer limit, @Bind("offset") Integer offset);
```

Comment:
We no longer need to apply `@Bind` to our method params, which is why they have been removed in this class. See #367